app/src/main/java/org/kiwix/kiwixmobile/webserver/wifi_hotspot/HotspotService.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.webserver.wifi_hotspot
+
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import android.widget.Toast
+import org.kiwix.kiwixmobile.KiwixApp
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.registerReceiver
+import org.kiwix.kiwixmobile.core.utils.ServerUtils.getSocketAddress
+import org.kiwix.kiwixmobile.webserver.WebServerHelper
+import org.kiwix.kiwixmobile.webserver.ZimHostCallbacks
+import org.kiwix.kiwixmobile.webserver.ZimHostFragment
+import javax.inject.Inject
+
+/**
+ * HotspotService is used to add a foreground service for the wifi hotspot.
+ * Created by Adeel Zafar on 07/01/2019.
+ */
+class HotspotService :
+  Service(),
+  IpAddressCallbacks,
+  HotspotStateReceiver.Callback {
+  @set:Inject
+  var webServerHelper: WebServerHelper? = null
+
+  @set:Inject
+  var hotspotNotificationManager: HotspotNotificationManager? = null
+
+  @set:Inject
+  var hotspotStateReceiver: HotspotStateReceiver? = null
+
+  private var zimHostCallbacks: ZimHostCallbacks? = null
+  private val serviceBinder: IBinder = HotspotBinder()
+
+  override fun onCreate() {
+    (this.application as KiwixApp).kiwixComponent
+      .serviceComponent()
+      .service(this)
+      .build()
+      .inject(this)
+    super.onCreate()
+    this.registerReceiver(hotspotStateReceiver!!)
```

Comment:
Remove double bang here. Wrap with `let` i.e `hotspotStateReceiver?.let(this::registerReceiver)`

Reply:
hi @gouri-panda  done